cmd/soroban-cli/src/commands/txn_result.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt::{Display, Formatter};
+
+use soroban_env_host::xdr::{Limits, Transaction, WriteXdr};
+
+pub enum TxnResult<R> {
+    Txn(Transaction),
```

Comment:
I pushed a change changing the `Xdr(String)` variant to a `Txn(Transaction)` variant. This is so that we don't have functions that return values of String or u32, or String or String, which is pretty confusing, and was I think the source of one bug (https://github.com/stellar/soroban-cli/pull/1267#discussion_r1591957598) that got fixed.

We don't need to eagerly convert the Transactions to XDR base64 strings, and that defers the conversion to string to the point where the code that cares most about the presentation layer can do so. It results in a little duplicated to_xdr_base64 code, but that seems really immaterial.

Also on its own the clarity gained as a reader feels like a worthwhile tradeoff.